evm-adapters/src/sputnik/cheatcodes/cheatcode_handler.rs

```diff
@@ -569,6 +572,24 @@ impl<'a, 'b, B: Backend, P: PrecompileSet> CheatcodeStackExecutor<'a, 'b, B, P>
                 // encode the data as Bytes
                 res = ethers::abi::encode(&[Token::Bytes(decoded.to_vec())]);
             }
+            HEVMCalls::GetCode(inner) => {
+                #[derive(Deserialize)]
+                struct ContractFile {
+                    bin: String,
+                }
+
+                self.add_debug(CheatOp::GETCODE);
+                let name = inner.0.replace(".sol", "");
+                let path = format!("./out/{}.sol/{}.json", name, name);
+                let mut file = File::open(path).unwrap();
+                let mut data = String::new();
+                file.read_to_string(&mut data).unwrap();
+
+                let contract_file: ContractFile = serde_json::from_str(&data).unwrap();
+                let code = hex::decode(contract_file.bin.replace("0x", "")).unwrap();
```

Comment:
Wonder if we can have a better way to get the artifact? This will only work if the artifact is under `./out/A.sol/A.json` which may not always be the case, e.g. in a hardhat style repository, or when there's >1 contracts in the file.

cc @mattsse not sure what the right way is here, given that we want to tell the EVM of how the directory structure looks like, but that's information that the compiler has, which is not part of the EVM infra

Reply:
i hate to say it but for this to be optimal, we need to grab the `out` folder from the `BuildArgs` and pass it all the way down... bleh

Reply:
This might not be the most efficient solution, but we could always just recurse through the entire directory tree and look for files of the form `ContractName.json`. This might get hairy though if we try and recurse through an entire `node_modules` folder though. Also, to handle files with multiple contracts in it, we could allow users to be more specific and write `getCode("ContractFile.sol:ContractName").

Reply:
> getCode("ContractFile.sol:ContractName").

I think this will be needed either way?

Let's brainstorm a bit more in the chat and see if we can come up w something that's a bit cleaner, vs pulling all BuildArgs up

Reply:
I think getcode should only for possible for contracts we previously compiled and have an artifact in `<root>/<artifacts>`. which we already read, so perhaps the right way to do this is to move all contracts/artifacts data into the `Backend` with like `Arc<Artifacts>` or something